skate-plugin/project-gen/src/jvmMain/kotlin/slack/tooling/projectgen/circuitgen/CircuitComponent.kt

```diff
@@ -0,0 +1,408 @@
+    /**
+     * Generate ViewModel class, e.g:
+     *
+     * @ContributesMultibinding( UserScope::class, boundType = ViewModel::class, )
+     * @ViewModelKey(${NAME}ViewModel::class) class ${NAME}ViewModel @Inject constructor(
+     *   slackDispatchers: SlackDispatchers, ) :
+     *   UdfViewModel<${NAME}Screen.State>(CloseableCoroutineScope.newMainScope(slackDispatchers)),
+     *   ${NAME}Screen.Events { private val state: MutableStateFlow<${NAME}Screen.State> =
+     *   MutableStateFlow(${NAME}Screen.State(events = this))
+     *
+     *   override fun state(): StateFlow<${NAME}Screen.State> = state }
+     */
+    val viewModel = ClassName(packageName, className)
+    val screenClass = ClassName(packageName, screenClassName(className))
+    val closeableCoroutineScope =
+      ClassName("slack.foundation.coroutines", "CloseableCoroutineScope")
+
+    val stateFun =
+      FunSpec.builder("state")
+        .addModifiers(KModifier.OVERRIDE)
+        .returns(STATE_FLOW.parameterizedBy(screenClass.nestedClass("State")))
+        .addStatement("return state")
+        .build()
+
+    val viewModelClass =
+      TypeSpec.classBuilder(viewModel)
+        .addAnnotation(
+          AnnotationSpec.builder(
+              ClassName("com.squareup.anvil.annotations", "ContributesMultibinding")
+            )
+            .apply { additionalScope.forEach { scope -> addMember("%T::class", scope) } }
+            .addMember("boundType = %T::class", VIEW_MODEL)
+            .build()
+        )
+        .addAnnotation(
+          AnnotationSpec.builder(VIEW_MODEL_KEY).addMember("%T::class", viewModel).build()
+        )
+        .primaryConstructor(
+          FunSpec.constructorBuilder()
+            .addParameter(ParameterSpec.builder("slackDispatchers", SLACK_DISPATCHER).build())
+            .addAnnotation(JAVA_INJECT)
+            .build()
+        )
+        .superclass(UDF_VIEW_MODEL.parameterizedBy(screenClass.nestedClass("State")))
+        .addSuperclassConstructorParameter(
+          "CloseableCoroutineScope.newMainScope(slackDispatchers)",
+          MemberName("slack.foundation.coroutines", "CloseableCoroutineScope.newMainScope"),
+        )
+        .addSuperinterface(screenClass.nestedClass("Events"))
+        .addProperty(
+          PropertySpec.builder(
+              "state",
+              MUTABLE_STATE_FLOW.parameterizedBy(screenClass.nestedClass("State")),
+            )
+            .initializer("MutableStateFlow(%T(events = this))", screenClass.nestedClass("State"))
+            .addModifiers(KModifier.PRIVATE)
+            .build()
+        )
+        .addFunction(stateFun)
+        .build()
+    return FileSpec.builder(packageName, className)
+      .addImport(closeableCoroutineScope.packageName, closeableCoroutineScope.simpleName)
+      .addType(viewModelClass)
+      .build()
+  }
+}
+
+data class AssistedInjectionConfig(val screen: Boolean = false, val navigator: Boolean = false)
```

Comment:
nit: in circuit's code gen we just model this with a list of `ParameterSpec`s, so it's extensible for future params we add too.

Reply:
tried to make this but might not make it in time before you left. mind if i do it as a follow up?